### ISSUE TITLE
left-pad "address" so they're sortable

### ISSUE THREAD

**Issue by willkg** [mozilla-services/minidump-stackwalk#30]
Currently, minidump-stackwalk emits a `crash_info` section with an `address` field that's the memory address where the process crashed. Values end up like `0x0`, `0x17fe7e4a`, `0x1b09319b441`, and so on.

It'd be more helpful for Crash Stats if this field was 0-padded. So then on a 32-bit system, `0x0` would be `0x00000000` and on a 64-bit system, `0x0` would be `0x0000000000000000`.

This would fix [bug 1032227](https://bugzilla.mozilla.org/show_bug.cgi?id=1032227) by making the addresses all the same length so they can be alphanumerically sorted.

**Comment by willkg** [comment]
On second thought, it's weird to left-pad this one field and not all the other memory address fields. I think I'm going to switch plans and fix this in the Socorro processor.

**Comment by willkg** [comment]
On third thought, maybe we should leftpad all the memory addresses. I asked on #crashreporting to see what others thought.

Jeff said it seems reasonable to leftpad memory addresses, but not offsets.

I asked @Gankra and @gabrielesvelto what they thought because whatever we do here, we're going to need to do in the new minidump-stackwalker.

Reopening this to figure that out.

**Comment by willkg** [comment]
```
char_width = (
    if cpu_arch == "x86":
        8
    else:
        16
)
```

We want to use the `char_width` to left-pad with 0 memory addresses that have fewer than `char_width` characters. Further, we want to truncate `FF` bytes for memory addresses with more than `char_width` characters.

Fields that should get padded:

* `crash_info.address`
* `largest_free_vm_block`
* `modules[].base_addr`
* `modules[].end_addr`
* `unloaded_modules[].base_addr`
* `unloaded_modules[].end_addr`

Example crash reports:

* x86 (32-bit): 2277001c-4153-4bac-8847-ea3170210519
* amd64 (64-bit): 9201dada-9b25-426d-8400-6916b0210519

Also, we need to update the schema in README when we make these changes.

Once I fix this, I can run it through a bunch of other crash report minidumps and see how it goes.

**Comment by willkg** [comment]
This was fixed in PR #36.
